### ISSUE TITLE
Question about leaf value update of multi-class objective (softmax)

### ISSUE THREAD

**Issue by lorentzenchr** [microsoft/LightGBM#6043]
## Description
In #3256, a new multiplicative factor `num_class_ / (num_class_ - 1)` was introduced for leaf node values for the multiclass objective (log loss). This factor is described in ["Greedy Function Approximation: A Gradient Boosting Machine" by Friedman](https://doi.org/10.1214/aos/1013203451).

In the end, this factor is the same as multiplying the learning rate. Nevertheless, I personally never understood its derivation (only ever a few words about it) and the following code snippet proves it wrong:

## Reproducible example
```python
import numpy as np


def prob(F):
    c = F.shape[1]
    p = np.exp(F) / np.sum(np.exp(F), axis=1)[:, None]
    return p

def loss(y, F):
    p = prob(F)
    return -np.sum(y * np.log(p))


y = np.array([[0, 1], [0, 1], [1, 0]])
F = np.array([[0.2, 0.8], [0.3, 0.7], [0.4, 0.6]])
p = prob(F)
K = y.shape[1]

f_no_scaling = (y - p) /(p * (1-p))
f_with_scaling = (K - 1)/K * f_no_scaling  # the factor is 1/2

loss(y, F + f_no_scaling), loss(y, F + f_with_scaling)
```
Result: `(0.06220676974920512, 0.353509963139608)`

So the one without scaling gives a smaller / better loss.

## Additional Comments
This is not really a bug, and maybe just needs clarification and documentation.

**Comment by guolinke** [comment]
@lorentzenchr you can refer to the page 357 in this paper: https://hastie.su.domains/Papers/AdditiveLogisticRegression/alr.pdf 
page

**Comment by lorentzenchr** [comment]
@guolinke Thanks for pointing to the literature. I really never got the sentence
> However, this procedure could be applied using any class as the base, not just the Jth. By averaging over all choices for the base class, we get the update

which justifies the factor $\frac{J-1}{J}$.
Because I never understood that, I tried it in code and my above snippet shows that this  factor worsens the loss.
